### PR TITLE
chore(meshes): makes MeshDetails injectable

### PR DESCRIPTION
**Pull request by kleinfreund**
Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: src/app/meshes/components/MeshDetails.vue [hunk 1]
```diff
@@ -1,166 +1,145 @@
...
+          class="columns"
+          style="--columns: 4"
+        >
+          <DefinitionCard>
+            <template #title>
+              {{ t('http.api.property.name') }}
+            </template>
+
+            <template #body>
+              <TextWithCopyButton :text="props.mesh.name" />
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/app/meshes/components/MeshDetails.vue:15]: `props.mesh` is also used to get the mTLS, logging, etc. data.

[the remark this answers is not in the document]
Comment (lahabana) [reply src/app/meshes/components/MeshDetails.vue:15]: Have we not talked about removing logging/trace etc? If yes then the only thing we have left is mTLS?

[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/app/meshes/components/MeshDetails.vue:15]: We're removing logging and tracing, but we're keeping mTLS and adding Zone Egress, yes. Both need access to `props.mesh`.

[the remark this answers is not in the document]
Comment (lahabana) [reply src/app/meshes/components/MeshDetails.vue:15]: Ok ideally it'd be nice to not have it like this but it's backend change.

**Review by lahabana — APPROVED** [review]

**Pull request merged**